File: equationsolver.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

// Ax = b

void mprint(double *A[], int n)
{
  int i, j;
  printf("[\n");
  for(i=0;i<n;i++)
    {
      for(j=0;j<n;j++)
	printf("%.2f ", A[i][j]);
      printf("| %.2f\n", A[i][n]);
    }
  printf("]\n");
}

int zero(double c)
{
  return -1e-15 <= c && c <= 1e-15;
}

void solv(double *A[], int n)
{
  int i, j, k;
  double *temp, f;
  for(i=0;i<n;i++)
    {
      for(k=i,j=i+1;j<n;j++)
        {
	  if(fabs(A[j][i]) > fabs(A[k][i]))
	    k = j;
        }
      if(zero(A[k][i]))
	{
	  for(j=0;j<n;A[j++][i]=0);
	  continue;
	}
      temp = A[k]; A[k] = A[i]; A[i] = temp;
      for(j=0;j<n;j++)
        {
	  if(i != j)
            {
	      f = zero(A[j][i]) ? 0 : A[j][i];
	      for(k=0;k<=n;k++)
		A[j][k] -= f * A[i][k] / A[i][i];
	      A[j][i] = 0;
            }
        }
    }
}

int main()
{
  int n, i, j, sol;
  double *A[100];
  for(i=0;i<100;i++)
    A[i] = malloc(sizeof(double)*101);
  while(1)
    {
      scanf("%d", &n);
      if(!n) break;
      for(i=0;i<n;i++)
        {
	  for(j=0;j<n;j++)
	    scanf("%lf", &A[i][j]);
        }
      for(i=0;i<n;i++)
	scanf("%lf", &A[i][n]);
      solv(A, n);
      sol = 1;
      for(i=0;i<n;i++)
        {
	  for(j=0; j<=n && zero(A[i][j]); j++);
	  if(j >= n)
	    sol = (j == n) ? 0 : 100;
        }
      if(sol != 1)
	printf("%s\n", (sol) ? "multiple" : "inconsistent");
      else
        {
	  for(i=0;i<n;i++)
	    printf("%f%c", A[i][n]/A[i][i], i<n-1?' ':'\n');
        }
    }
  for(i=0;i<100;free(A[i++]));
}

File: equationsolver.py
# Ax = b

def zero(c):
    return -1e-15 <= c <= 1e-15

def solv(A, n):
    for i in range(n):
        k = i
        for j in range(i+1, n):
            if abs(A[j][i]) > abs(A[k][i]): k = j;
        if zero(A[k][i]):
            for j in range(n):
                A[j][i] = 0
            continue
        A[i], A[k] = A[k], A[i]
        for j in range(n):
            if i != j:
                f = A[j][i] if not zero(A[j][i]) else 0
                for k in range(n+1):
                    A[j][k] -= f * A[i][k] / A[i][i]
                    if k == i:
                        A[j][k] = 0.0

while True:
    n = int(input())
    if not n: break;
    A = [[float(_) for _ in input().split()] for _ in range(n)]
    b = [float(_) for _ in input().split()]
    A = [A[i] + [b[i]] for i in range(n)]
    solv(A, n)
    sol = 1
    for i in range(n):
        j = 0
        while j <= n and zero(A[i][j]):
            j += 1
        if j >= n:
            sol = 0 if j == n else 100
    if sol == 1:
        x = [A[i][-1]/A[i][i] for i in range(n)]
        print(' '.join([str(_) for _ in x]))
    else:
        print('inconsistent' if not sol else 'multiple')
